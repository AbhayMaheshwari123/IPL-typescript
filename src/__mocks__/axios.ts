export default {
    get:jest.fn().mockResolvedValueOnce({data:{}}),
    interceptors:{
        request:{
            use:jest.fn()
        },
        response:{
            use:jest.fn()
        }
    }
}