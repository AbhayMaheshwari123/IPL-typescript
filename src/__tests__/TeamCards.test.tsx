import { render, screen, fireEvent } from "@testing-library/react";
import axios from "axios"
import Cards from "../Components/TeamCards";

const mockedAxios = axios as jest.Mocked<typeof axios>
it("Teams name check", async () => {
  mockedAxios.get.mockResolvedValueOnce({
    data: [
      {
        "id": "chennai-super-kings",
        "teamName": "Chennai Super Kings",
        "winningYears": [
          2010,
          2011,
          2018
        ],
        "venue": "M. A. Chidambaram Stadium"
      },
      {
        "id": "delhi-capitals",
        "teamName": "Delhi Capitals",
        "winningYears": [],
        "venue": "Feroz Shah Kotla Ground"
      },
      {
        "id": "kings-xi-punjab",
        "teamName": "Kings XI Punjab",
        "winningYears": [],
        "venue": "IS Bindra Stadium"
      },
      {
        "id": "kolkata-knight-riders",
        "teamName": "Kolkata Knight Riders",
        "winningYears": [
          2012,
          2014
        ],
        "venue": "Eden Gardens"
      },
      {
        "id": "mumbai-indians",
        "teamName": "Mumbai Indians",
        "winningYears": [
          2013,
          2015,
          2017,
          2019
        ],
        "venue": "Wankhede Stadium"
      },
      {
        "id": "rajasthan-royals",
        "teamName": "Rajasthan Royals",
        "winningYears": [
          2008
        ],
        "venue": "Sawai Mansingh Stadium"
      },
      {
        "id": "royal-challengers-bangalore",
        "teamName": "Royal Challengers Bangalore",
        "winningYears": [],
        "venue": "M. Chinnaswamy Stadium"
      },
      {
        "id": "sunrisers-hyderabad",
        "teamName": "Sunrisers Hyderabad",
        "winningYears": [
          2016
        ],
        "venue": "Rajiv Gandhi Intl. Cricket Stadium"
      }
    ]
  });
  render(<Cards />);
  const cards = await screen.findAllByTestId('cards');
  expect(cards[0]).toHaveTextContent('Chennai Super Kings');
  expect(cards[1]).toHaveTextContent('Delhi Capitals');
  expect(cards[2]).toHaveTextContent('Kings XI Punjab');
  expect(cards[3]).toHaveTextContent('Kolkata Knight Riders');
  expect(cards[4]).toHaveTextContent('Mumbai Indians');
  expect(cards[5]).toHaveTextContent('Rajasthan Royals');
  expect(cards[6]).toHaveTextContent('Royal Challengers Bangalore');
  expect(cards[7]).toHaveTextContent('Sunrisers Hyderabad');

})

it("Venue name check", async () => {
  mockedAxios.get.mockResolvedValueOnce({
    data: [
      {
        "id": "chennai-super-kings",
        "teamName": "Chennai Super Kings",
        "winningYears": [
          2010,
          2011,
          2018
        ],
        "venue": "M. A. Chidambaram Stadium"
      },
      {
        "id": "delhi-capitals",
        "teamName": "Delhi Capitals",
        "winningYears": [],
        "venue": "Feroz Shah Kotla Ground"
      },
      {
        "id": "kings-xi-punjab",
        "teamName": "Kings XI Punjab",
        "winningYears": [],
        "venue": "IS Bindra Stadium"
      },
      {
        "id": "kolkata-knight-riders",
        "teamName": "Kolkata Knight Riders",
        "winningYears": [
          2012,
          2014
        ],
        "venue": "Eden Gardens"
      },
      {
        "id": "mumbai-indians",
        "teamName": "Mumbai Indians",
        "winningYears": [
          2013,
          2015,
          2017,
          2019
        ],
        "venue": "Wankhede Stadium"
      },
      {
        "id": "rajasthan-royals",
        "teamName": "Rajasthan Royals",
        "winningYears": [
          2008
        ],
        "venue": "Sawai Mansingh Stadium"
      },
      {
        "id": "royal-challengers-bangalore",
        "teamName": "Royal Challengers Bangalore",
        "winningYears": [],
        "venue": "M. Chinnaswamy Stadium"
      },
      {
        "id": "sunrisers-hyderabad",
        "teamName": "Sunrisers Hyderabad",
        "winningYears": [
          2016
        ],
        "venue": "Rajiv Gandhi Intl. Cricket Stadium"
      }
    ]
  })
  render(<Cards />)
  const cards = await screen.findAllByTestId('cards');
  expect(cards[0]).toHaveTextContent('M. A. Chidambaram Stadium');
  expect(cards[1]).toHaveTextContent('Feroz Shah Kotla Ground');
  expect(cards[2]).toHaveTextContent('IS Bindra Stadium');
  expect(cards[3]).toHaveTextContent('Eden Gardens');
  expect(cards[4]).toHaveTextContent('Wankhede Stadium');
  expect(cards[5]).toHaveTextContent('Sawai Mansingh Stadium');
  expect(cards[6]).toHaveTextContent('M. Chinnaswamy Stadium');
  expect(cards[7]).toHaveTextContent('Rajiv Gandhi Intl. Cricket Stadium');

});

it("Winning Years check", async () => {
  mockedAxios.get.mockResolvedValueOnce({
    data: [
      {
        "id": "chennai-super-kings",
        "teamName": "Chennai Super Kings",
        "winningYears": [
          2010,
          2011,
          2018
        ],
        "venue": "M. A. Chidambaram Stadium"
      },
      {
        "id": "delhi-capitals",
        "teamName": "Delhi Capitals",
        "winningYears": [],
        "venue": "Feroz Shah Kotla Ground"
      },
      {
        "id": "kings-xi-punjab",
        "teamName": "Kings XI Punjab",
        "winningYears": [],
        "venue": "IS Bindra Stadium"
      },
      {
        "id": "kolkata-knight-riders",
        "teamName": "Kolkata Knight Riders",
        "winningYears": [
          2012,
          2014
        ],
        "venue": "Eden Gardens"
      },
      {
        "id": "mumbai-indians",
        "teamName": "Mumbai Indians",
        "winningYears": [
          2013,
          2015,
          2017,
          2019
        ],
        "venue": "Wankhede Stadium"
      },
      {
        "id": "rajasthan-royals",
        "teamName": "Rajasthan Royals",
        "winningYears": [
          2008
        ],
        "venue": "Sawai Mansingh Stadium"
      },
      {
        "id": "royal-challengers-bangalore",
        "teamName": "Royal Challengers Bangalore",
        "winningYears": [],
        "venue": "M. Chinnaswamy Stadium"
      },
      {
        "id": "sunrisers-hyderabad",
        "teamName": "Sunrisers Hyderabad",
        "winningYears": [
          2016
        ],
        "venue": "Rajiv Gandhi Intl. Cricket Stadium"
      }
    ]
  })
  render(<Cards />)
  const cards = await screen.findAllByTestId('cards');
  expect(cards[0]).toHaveTextContent('2010 2011 2018');
  expect(cards[3]).toHaveTextContent('2012 2014');
  expect(cards[4]).toHaveTextContent('2013 2015 2017 2019');
  expect(cards[5]).toHaveTextContent('2008');
  expect(cards[7]).toHaveTextContent('2016');

});

it("Cards Snapshot Testing", async () => {
  mockedAxios.get.mockResolvedValueOnce({
    data: [
      {
        "id": "chennai-super-kings",
        "teamName": "Chennai Super Kings",
        "winningYears": [
          2010,
          2011,
          2018
        ],
        "venue": "M. A. Chidambaram Stadium"
      },
      {
        "id": "delhi-capitals",
        "teamName": "Delhi Capitals",
        "winningYears": [],
        "venue": "Feroz Shah Kotla Ground"
      },
      {
        "id": "kings-xi-punjab",
        "teamName": "Kings XI Punjab",
        "winningYears": [],
        "venue": "IS Bindra Stadium"
      },
      {
        "id": "kolkata-knight-riders",
        "teamName": "Kolkata Knight Riders",
        "winningYears": [
          2012,
          2014
        ],
        "venue": "Eden Gardens"
      },
      {
        "id": "mumbai-indians",
        "teamName": "Mumbai Indians",
        "winningYears": [
          2013,
          2015,
          2017,
          2019
        ],
        "venue": "Wankhede Stadium"
      },
      {
        "id": "rajasthan-royals",
        "teamName": "Rajasthan Royals",
        "winningYears": [
          2008
        ],
        "venue": "Sawai Mansingh Stadium"
      },
      {
        "id": "royal-challengers-bangalore",
        "teamName": "Royal Challengers Bangalore",
        "winningYears": [],
        "venue": "M. Chinnaswamy Stadium"
      },
      {
        "id": "sunrisers-hyderabad",
        "teamName": "Sunrisers Hyderabad",
        "winningYears": [
          2016
        ],
        "venue": "Rajiv Gandhi Intl. Cricket Stadium"
      }
    ]
  });
  render(<Cards />);
  const cards = await screen.findAllByTestId('cards');
  for (const i of cards) {
    expect(i).toMatchSnapshot()
  }

})

// it("clickhandler", () => {
//   render(<Cards />)
//   const handleclick = jest.fn()
//   fireEvent.click(screen.getByTestId('cards'))
//   expect(handleclick).toHaveBeenCalledTimes(1)
// })
