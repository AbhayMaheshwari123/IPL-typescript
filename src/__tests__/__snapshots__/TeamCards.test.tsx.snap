// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards 1`] = `null`;

exports[`Cards Snapshot Testing 1`] = `
<div
  class="teamcard-0-2-2 chennai-super-kings-0-2-8"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Chennai Super Kings
    </h3>
    <h6
      class="venue-0-2-4"
    >
      M. A. Chidambaram Stadium
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2010 
        2011 
        2018 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Cards Snapshot Testing 2`] = `
<div
  class="teamcard-0-2-2 delhi-capitals-0-2-9"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Delhi Capitals
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Feroz Shah Kotla Ground
    </h6>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Cards Snapshot Testing 3`] = `
<div
  class="teamcard-0-2-2 kings-xi-punjab-0-2-12"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Kings XI Punjab
    </h3>
    <h6
      class="venue-0-2-4"
    >
      IS Bindra Stadium
    </h6>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Cards Snapshot Testing 4`] = `
<div
  class="teamcard-0-2-2 kolkata-knight-riders-0-2-15"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Kolkata Knight Riders
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Eden Gardens
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2012 
        2014 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Cards Snapshot Testing 5`] = `
<div
  class="teamcard-0-2-2 mumbai-indians-0-2-10"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Mumbai Indians
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Wankhede Stadium
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2013 
        2015 
        2017 
        2019 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Cards Snapshot Testing 6`] = `
<div
  class="teamcard-0-2-2 rajasthan-royals-0-2-11"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Rajasthan Royals
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Sawai Mansingh Stadium
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2008 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Cards Snapshot Testing 7`] = `
<div
  class="teamcard-0-2-2 royal-challengers-bangalore-0-2-13"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Royal Challengers Bangalore
    </h3>
    <h6
      class="venue-0-2-4"
    >
      M. Chinnaswamy Stadium
    </h6>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Cards Snapshot Testing 8`] = `
<div
  class="teamcard-0-2-2 sunrisers-hyderabad-0-2-14"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Sunrisers Hyderabad
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Rajiv Gandhi Intl. Cricket Stadium
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2016 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Teams name check 1`] = `
<div
  class="teamcard-0-2-2 chennai-super-kings-0-2-8"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Chennai Super Kings
    </h3>
    <h6
      class="venue-0-2-4"
    >
      M. A. Chidambaram Stadium
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2010 
        2011 
        2018 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Teams name check 2`] = `
<div
  class="teamcard-0-2-2 delhi-capitals-0-2-9"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Delhi Capitals
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Feroz Shah Kotla Ground
    </h6>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Teams name check 3`] = `
<div
  class="teamcard-0-2-2 kings-xi-punjab-0-2-12"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Kings XI Punjab
    </h3>
    <h6
      class="venue-0-2-4"
    >
      IS Bindra Stadium
    </h6>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Teams name check 4`] = `
<div
  class="teamcard-0-2-2 kolkata-knight-riders-0-2-15"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Kolkata Knight Riders
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Eden Gardens
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2012 
        2014 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Teams name check 5`] = `
<div
  class="teamcard-0-2-2 mumbai-indians-0-2-10"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Mumbai Indians
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Wankhede Stadium
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2013 
        2015 
        2017 
        2019 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Teams name check 6`] = `
<div
  class="teamcard-0-2-2 rajasthan-royals-0-2-11"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Rajasthan Royals
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Sawai Mansingh Stadium
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2008 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Teams name check 7`] = `
<div
  class="teamcard-0-2-2 royal-challengers-bangalore-0-2-13"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Royal Challengers Bangalore
    </h3>
    <h6
      class="venue-0-2-4"
    >
      M. Chinnaswamy Stadium
    </h6>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;

exports[`Teams name check 8`] = `
<div
  class="teamcard-0-2-2 sunrisers-hyderabad-0-2-14"
  data-testid="cards"
>
  <img
    alt="Team Logo"
    class="logo-0-2-5"
  />
  <div>
    <h3
      class="teamname-0-2-6"
    >
      Sunrisers Hyderabad
    </h3>
    <h6
      class="venue-0-2-4"
    >
      Rajiv Gandhi Intl. Cricket Stadium
    </h6>
    <div>
      <h4
        class="win-0-2-7"
      >
        2016 
      </h4>
    </div>
  </div>
  <h5
    class="home-0-2-3"
  >
    Team Home
  </h5>
</div>
`;
